File: fibonacci.py
# Python --
# Demostración de recursividad, uso de operador ternario y convertir lista en cadena de caracteres.

def fibonacci(n):
    return fibonacci(n-1) + fibonacci(n-2) if n > 1 else n
    
resultado = []

for i in range(10):
    resultado.append(fibonacci(i))

print(' '.join(map(str, resultado)))

File: fibonacci.rs
// Rust --
// Demostración de recursividad, uso de match y mostrar contenido de un vector.

fn fibonacci(n: u8) -> u8 {
    match n {
      0 | 1 => n,
      _ => fibonacci(n-1) + fibonacci(n-2),
    }
  }
  
fn main() {
    let mut resultado = vec![];

    for i in 0..10 {
    resultado.push(fibonacci(i));
    }

    for valor in resultado.iter() {
    print!("{} ", valor);
    }
}